### PR TITLE
Target typed new

### PR DESCRIPTION
**Pull request by yufeih**
[AB#347528](https://dev.azure.com/ceapex/Engineering/_workitems/edit/347528/)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6895)

### CODE REVIEW

**Comment by yufeih** [comment]
The actual build has succeeded: https://dev.azure.com/ceapex/Engineering/_build/results?buildId=251570&view=results
Somehow GitHub does not show it.

File: src/docfx/config/ops/OpsAccessor.cs [hunk 1]
```diff
@@ -30,25 +30,25 @@ internal class OpsAccessor : ILearnServiceAccessor
...
-        private static readonly Lazy<Task<string>> s_opsTokenProd = new Lazy<Task<string>>(() => GetSecret("OpsBuildTokenProd"));
-        private static readonly Lazy<Task<string>> s_opsTokenSandbox = new Lazy<Task<string>>(() => GetSecret("OpsBuildTokenSandbox"));
+        private static readonly SecretClient s_secretClient = new(new("https://docfx.vault.azure.net"), new DefaultAzureCredential());
+        private static readonly Lazy<Task<string>> s_opsTokenProd = new(() => GetSecret("OpsBuildTokenProd"));
+        private static readonly Lazy<Task<string>> s_opsTokenSandbox = new(() => GetSecret("OpsBuildTokenSandbox"));
 
         private readonly Action<HttpRequestMessage> _credentialProvider;
         private readonly ErrorBuilder _errors;
-        private readonly HttpClient _http = new HttpClient();
+        private readonly HttpClient _http = new();
```
Comment (live1206) [inline src/docfx/config/ops/OpsAccessor.cs:51]: why not `static` as well for these fields?

File: src/docfx/lib/msgraph/MicrosoftGraphAccessor.cs [hunk 2]
```diff
@@ -14,27 +14,22 @@ namespace Microsoft.Docs.Build
 {
     internal class MicrosoftGraphAccessor
     {
-        private readonly IGraphServiceClient? _msGraphClient;
+        private readonly GraphServiceClient? _msGraphClient;
```
Comment (live1206) [inline src/docfx/lib/msgraph/MicrosoftGraphAccessor.cs:17]: Do we need to replace all interfaces with class in order to use `new()`?

**Review by live1206 — APPROVED** [review]

**Comment by herohua** [comment]
Some of the instances are changed from `IReadOnlyDictionary<TKey, TValue>` to `Dictionary<TKey, TValue>` - would this be a concern? @yufeih @live1206

**Comment by yufeih** [comment]
> Some of the instances are changed from `IReadOnlyDictionary<TKey, TValue>` to `Dictionary<TKey, TValue>` - would this be a concern? @yufeih @live1206

Yes, and it'll be addressed separately, there are many places that does not enforce readonlyness, and we'll probably need an analyzer to detect and warn such cases.

**Pull request merged**